### PR TITLE
New sample: receiving authenticated requests

### PR DESCRIPTION
**Pull request by glasnt**
## Description

Fixes b/128426240

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: run/service-auth/receive_test.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            "--async",
+            "--platform=managed",
+            "--region=us-central1",
+            "--quiet",
+        ],
+        check=True,
+    )
+
+
+def test_auth(services):
```
Comment (dandhlee) [inline run/service-auth/receive_test.py:94]: Currently it only tests for `if auth_type.lower() == "bearer":` portion, are there plans to cover for the rest of the cases?

Comment (glasnt) [reply run/service-auth/receive_test.py:94]: I wanted to show consuming auth tokens; I have a proof of concept that also tests `basic`, but I planned on only showing to consume things we already show how to send.

Comment (dandhlee) [reply run/service-auth/receive_test.py:94]: Sounds good. Could you open an issue to keep track and handle for those eventually?

**Review by dandhlee — APPROVED** [review]

File: run/service-auth/receive_test.py (hunk 1 above)
Comment (glasnt) [reply run/service-auth/receive_test.py:94]: We don't have samples that show sending these (we prefer tokens over passwords) so I don't plan to show how to receive these

**Review by averikitsch — APPROVED** [review]

File: run/service-auth/Procfile [hunk 2]
```diff
@@ -0,0 +1 @@
+web: gunicorn app:app --bind 0.0.0.0:$PORT
```
[the remark this answers is not in the document]
Comment (glasnt) [reply run/service-auth/Procfile:1]: I'll copy the best practice from our samples here.

File: run/service-auth/receive.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    if auth_header:
+        auth_type, creds = auth_header.split(" ", 1)
+
+        if auth_type.lower() == "bearer":
+            claims = jwt.decode(creds, verify=False)
+            return f"Hello {claims['email']}!\n"
+
+        else:
+            return f"Unhandled header format ({auth_type}).\n"
+    return "Anonymous user.\n"
```
[the remark this answers is not in the document]
Comment (glasnt) [reply run/service-auth/receive.py:43]: I don't think we should, I think it's more useful to return successfully in this instance.

**Pull request merged**